File: bindings/nodejs/examples/binding_examples/2-generate-address.js
/**
 * This example generates a new address.
 */

require('dotenv').config();

async function run() {
    const { AccountManager } = require('@iota/wallet');
    const manager = new AccountManager({
        storagePath: './alice-database',
    });

    manager.setStrongholdPassword(process.env.SH_PASSWORD);

    const account = manager.getAccount('Alice');
    console.log('Account:', account.alias());

    // Always sync before doing anything with the account
    await account.sync();
    console.log('Syncing...');

    const address = account.generateAddress();
    console.log('New address:', address);

    // You can also get the latest unused address:
    const addressObject = account.latestAddress();
    console.log('Address:', addressObject.address);

    // Use the Chrysalis Faucet to send testnet tokens to your address:
    console.log(
        'Fill your address with the Faucet: https://faucet.chrysalis-devnet.iota.cafe/',
    );

    const addresses = account.listAddresses();
    console.log('Addresses:', addresses);
}

run();

File: bindings/python/native/example/2-generate-address.py
from iota_wallet import IotaWallet

# This example genrates a new address.

client_options = {
    'primaryNode': {'url': 'https://api.lb-0.h.chrysalis-devnet.iota.cafe'},
    'localPow': True,
}
signer = ("flame fever pig forward exact dash body idea link scrub tennis minute " +
          "surge unaware prosper over waste kitten ceiling human knife arch situate civil")

wallet = IotaWallet('./alice-database', client_options, signer)
account = wallet.get_account('Alice')
print(f'Account: {account}')

# Always sync before doing anything with the account
response = account.sync_account()
# response = account.sync_account({'syncAllAddresses': True})
print(f'Syncing... {response}')

addresses = account.generate_addresses(5)
# addresses = account.generate_addresses(
#     2, {'internal': True, 'metadata': {'syncing': True, 'network': 'Testnet'}})
print(f'Addresses: {addresses}')
